<launch>
    <arg name="bundle_filenames" />
    
    <arg name="image_topic" />
    <arg name="info_topic" />
    <arg name="pointcloud_topic"  />

    <arg name="output_frame" default="" />

    <arg name="marker_size" default="7.0" />

    <arg name="max_new_marker_error" default="0.2" />
    <arg name="max_track_error" default="0.2" />

    <arg name="publish_marker_tf" default="False" />

    <arg name="median_filter_size" default="10" />

    <arg name="use_depth" default="False" />
  
    <arg name="use_ransac" default="False" />
    <arg name="ransac_use_corners" default="True" />
    <arg name="ransac_inlier_threshold" default="0.02" />
    <arg name="ransac_max_iterations" default="1000" />
    <arg name="ransac_refine" default="True" />
    <arg name="ransac_dense_spacing" default="0.0" />

    <arg name="rviz" default="False" />
    
    <include file="$(find openni_launch)/launch/openni.launch"/>
    

    <node name="bundle_tracker" pkg="ar_tracking_tools" type="alvarBundleTracker" respawn="false" output="screen" >
        <param name="bundle_filenames" type="string" value="$(arg bundle_filenames)" />
        
        <param name="publish_marker_tf" type="bool" value="$(arg publish_marker_tf)" />
        <param name="max_new_marker_error" type="double" value="$(arg max_new_marker_error)" />
        <param name="max_track_error" type="double" value="$(arg max_track_error)" />
        
        <param name="use_depth" type="bool" value="$(arg use_depth)" />

        <param name="median_filter_size" type="double" value="$(arg median_filter_size)" />
        
        <param name="use_ransac" type="bool" value="$(arg use_ransac)" />
        <param name="ransac_use_corners" type="bool" value="$(arg ransac_use_corners)" />
        <param name="ransac_inlier_threshold" type="double" value="$(arg ransac_inlier_threshold)" />
        <param name="ransac_max_iterations" type="int" value="$(arg ransac_max_iterations)" />
        <param name="ransac_refine" type="bool" value="$(arg ransac_refine)" />
        <param name="ransac_dense_spacing" type="double" value="$(arg ransac_dense_spacing)" />

        <remap from="in_image" to="$(arg image_topic)" />
        <remap from="in_cloud" to="$(arg pointcloud_topic)" />
        <remap from="in_camera_info" to="$(arg info_topic)" />
    </node>
</launch>
